Drop PHP 7 support, add PHP 8.2 support

weierophinney opened the pull request:
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | no
| QA            | yes

### Description

This patch adds support for PHP 8.2, while dropping support for PHP 7. Additionally, it bumps the laminas-servicemanager dependency to 3.11, dropping support for container-interop.

weierophinney committed:
feat: bump to PHP 8 series only

- Remove support for PHP 7
- Add support for PHP 8.2
- Bump laminas-api-tools deps to versions that work with 8.2

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
feat: bump to laminas-servicemanager 3.11+

First release removing container-interop.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
qa: switch to `$varname::class` usage instead of `get_class($varname)`

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
qa: resolve Psalm errors, and stuff what we cannot into baseline

Makes a few properties nullable, as they were operating that way already.
Stuffs remaining issues into baseline.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
qa: use local phpunit schema

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
qa: bump minimum PHPUnit version to 9.5.10

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney commented:
Blocked by laminas-api-tools/api-tools-oauth2#42.

weierophinney committed:
qa: bump api-tools-oauth2 to 1.9

Picks up PHP 8.2 support.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

gsteel reviewed (approved):
I see the upstream types are out of whack in laminas-authentication, so the psalm issues should go in the baseline, but other than that 👍

weierophinney commented:
Failures are from Psalm, and I cannot reproduce them locally, even with the `--no-cache` flag.

From what I can tell, `Laminas\ApiTools\Authentication\HttpAdapter::provides()` is calling on its composed `Laminas\Authentication\Adapter\Http` instance to determine if either of a basic or digest provider is available, using the `getBasicResolver()` and `getDigestResolver()` respectively.

Psalm is complaining that when calling these methods, we are checking to see if `null` is returned, claiming that neither method is nullable. However, this is not true; in the locked version used, 2.13.0, these each are nullable methods, so the checks are correct.

Since I cannot reproduce, I'm not sure how to exclude these in order to resolve the issue.

**Update**: I'd made changes to the laminas-authentication libraries locally, which is why I could not recreate. Resolved now.

weierophinney committed:
qa: stuff some errors into psalm baseline

Laminas authentication resolvers are nullable, but not explicitly so currently.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

Ocramius reviewed (approved):
LGTM :+1:
